### PR TITLE
Prepare heroku deployment.

### LINKED ISSUE #376: Deploy http service somewhere
Either Heroku, or GCP, or AWS.
Depends on the task-boot issue to support deploying to one of those services.

### PR DESCRIPTION
**Pull request by Lothiraldan**
Fixes #376.

### CODE REVIEW

File: .taskcluster.yml [hunk 1]
```diff
@@ -224,6 +224,42 @@ tasks:
             owner: ${user}@users.noreply.github.com
             source: ${repository}/raw/${head_rev}/.taskcluster.yml
 
+      - $if: 'tasks_for == "github-push" && head_branch[:10] == "refs/tags/"'
```
Comment (marco-c) [inline .taskcluster.yml]: We should deploy at the end of the data pipeline

File: http_service/Dockerfile [hunk 2]
```diff
@@ -18,4 +18,4 @@ ENV CHECK_MODELS="${CHECK_MODELS}"
 
 RUN python /code/bugbug_http_service/check_models.py
 
-CMD ["gunicorn", "-b", "0.0.0.0:8000", "bugbug_http_service.app", "--preload", "--timeout", "30", "-w", "3"]
+CMD gunicorn -b 0.0.0.0:$PORT bugbug_http_service.app --preload --timeout 30 -w 3
```
Comment (marco-c) [inline http_service/Dockerfile]: Is the other format not OK on HEroku?

File: http_service/models.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+LOGGER = logging.getLogger()
+
+MODELS = {
+    "defectenhancementtask": DefectEnhancementTaskModel,
+    "component": ComponentModel,
+    "regression": RegressionModel,
+}
+MODELS_NAMES = MODELS.keys()
+MODELS_DIR = os.path.join(os.path.dirname(__file__), "models")
+BASE_URL = "https://index.taskcluster.net/v1/task/project.releng.services.project.testing.bugbug_train.latest/artifacts/public"
```
Comment (marco-c) [inline http_service/models.py]: We should use the new URL

File: http_service/models.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import requests
+
+from bugbug.models.component import ComponentModel
+from bugbug.models.defect_enhancement_task import DefectEnhancementTaskModel
+from bugbug.models.regression import RegressionModel
+
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger()
+
+MODELS = {
```
Comment (marco-c) [inline http_service/models.py]: I wonder if we couldn't reuse at least part of the central place where we define models

File: .taskcluster.yml (hunk 1 above)
Comment (Lothiraldan) [reply .taskcluster.yml]: If we do that, that means we need to wait for a full data-pipeline in order to deploy any code change, including a hotfix, that's sub-obtimal. But we are generating an HTTP Service Docker image without the models so it's effectively broken.

I'm not fan of either solutions, @marco-c what do you think?

File: http_service/Dockerfile (hunk 2 above)
Comment (Lothiraldan) [reply http_service/Dockerfile]: No the other format couldn't be updated as the list-based format is directly executed by Docker. This format means that it's wrapped by bash for argument parsing and the nice side-effect is that bash is doing the variable interpolation for us.

File: http_service/models.py (hunk 4 above)
Comment (Lothiraldan) [reply http_service/models.py]: Yes now that we have merged the centralization PR, we should, let me update this PR

File: .taskcluster.yml (hunk 1 above)
Comment (marco-c) [reply .taskcluster.yml]: For now we have to do the non-broken thing, we can figure out how to deploy fixes later (file an issue so we don't forget).

Comment (Lothiraldan) [reply .taskcluster.yml]: Done https://github.com/mozilla/bugbug/issues/486

File: http_service/models.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+from urllib.request import urlretrieve
+
+import requests
+
+from bugbug.models import load_model as bugbug_load_model
+
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger()
+
+MODELS_NAMES = ["defectenhancementtask", "component", "regression"]
```
Comment (marco-c) [inline http_service/models.py]: We need a better way here, so we can avoid hardcoding the models. Maybe we could parse the yml file to check what we are training? Let's file an issue so we don't forget.

**Pull request merged**